pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/gates/tests/Test_ExaTNCuda_NonParam.cpp

```diff
@@ -33,6 +33,371 @@ namespace cuUtil = Pennylane::LightningGPU::Util;
 } // namespace
 /// @endcond
 
+TEMPLATE_TEST_CASE("ExaTNCuda::Gates::Identity", "[ExaTNCuda_Nonparam]", float,
+                   double) {
+    const bool inverse = GENERATE(false, true);
+    {
+        using cp_t = std::complex<TestType>;
+        std::size_t num_qubits = 3;
+        DevTag<int> dev_tag{0, 0};
+
+        SECTION("Apply different wire indices") {
+            const std::size_t index = GENERATE(0, 1, 2);
+            ExaTNCuda<TestType> exatn_state{num_qubits, dev_tag};
+
+            exatn_state.applyOperation("Hadamard", {index}, inverse);
+
+            exatn_state.applyOperation("Identity", {index}, inverse);
+            cp_t expected(1.0 / std::sqrt(2), 0);
+
+            auto results = exatn_state.getDataVector();
+
+            CHECK(expected.real() ==
+                  Approx(results[0b1 << ((num_qubits - 1 - index))].real()));
+            CHECK(expected.imag() ==
+                  Approx(results[0b1 << ((num_qubits - index - 1))].imag()));
```

Comment:
Could you please try using our in-house approximations to compare these complex numbers?